projects/ng-components/lib/toggle-selector/toggle-selector.component.spec.ts

```diff
@@ -0,0 +1,172 @@
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { ToggleSelectorComponent, IToggleSelector } from "./toggle-selector.component";
+import { Component, DebugElement, ViewChild } from "@angular/core";
+import { FormsModule } from "@angular/forms";
+import { CommonModule } from "@angular/common";
+import { By } from "@angular/platform-browser";
+
+@Component({
+    selector: "test-sebng-toggle-selector",
+    template: `<sebng-toggle-selector
+        #toggleSelector
+        [multi]="multi"
+        [list]="list"
+        [(ngModel)]="model"
+        [disabled]="disabled"
+        [error]="error"
+        [errorMessage]="errorMessage"
+    ></sebng-toggle-selector>`,
+})
+class ToggleSelectorTestComponent {
+    @ViewChild("toggleSelector") toggleSelector: ToggleSelectorComponent;
+    list: Array<IToggleSelector>;
+    multi: boolean = false;
+    disabled?: boolean = false;
+    model?: any;
+    error: boolean;
+    errorMessage: string;
+
+    constructor() {
+        this.list = [
+            { value: "1", label: "Four mississipi", disabled: true },
+            { value: "2", label: "Five mississipi" },
+        ];
+    }
+}
+
+describe("ToggleSelectorComponent", () => {
+    let component: ToggleSelectorTestComponent;
+    let fixture: ComponentFixture<ToggleSelectorTestComponent>;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [FormsModule, CommonModule],
+            declarations: [ToggleSelectorComponent, ToggleSelectorTestComponent],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ToggleSelectorTestComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    it("Testing two-way data binding", async () => {
+        component.model = component.list[1];
+
+        const ngModelChange = spyOn(component.toggleSelector, "writeValue").and.callThrough();
+        const generateDisplayArray = spyOn(component.toggleSelector, "generateCheckedItems");
+        fixture.detectChanges();
+        await fixture.whenStable().then(() => {
+            expect(ngModelChange).toHaveBeenCalled();
+            expect(generateDisplayArray).toHaveBeenCalled();
+        });
+    });
+
+    it("Should disable the element when disabled prop is set to true", () => {
+        component.disabled = true;
+        fixture.detectChanges();
+        const container: DebugElement = fixture.debugElement.query(By.css(".disabled"));
+        const elements: DebugElement[] = fixture.debugElement.queryAll(By.css(".toggle-selector input:disabled"));
+        expect(container).toBeDefined();
+        expect(container).not.toBeNull();
+        expect(elements).toBeDefined();
+        expect(elements).toBeTruthy();
+        expect(elements.length).toBe(component.list.length);
```

Comment:
```suggestion
        expect(container).not.toBeNull();
        expect(elements).toHaveLength(component.list.length);
```

Reply:
This throws an error Property 'toHaveLength' does not exist on type 'ArrayLikeMatchers'